x-pack/solutions/security/packages/kbn-cloud-security-posture/public/src/components/fleet_extensions/aws_credentials_form/aws_credentials_form.test.tsx

```diff
@@ -0,0 +1,458 @@
+    });
+  });
+
+  describe('Event Handlers', () => {
+    it('can switch between CloudFormation and Manual setup formats', () => {
+      renderWithProviders(defaultProps);
+
+      const cloudFormationRadio = screen.getByTestId(
+        AWS_CREDENTIALS_TYPE_OPTIONS_TEST_SUBJECTS.CLOUDFORMATION
+      );
+      const manualRadio = screen.getByTestId(AWS_CREDENTIALS_TYPE_OPTIONS_TEST_SUBJECTS.MANUAL);
+
+      // Should be able to interact with radio buttons
+      expect(cloudFormationRadio).not.toBeDisabled();
+      expect(manualRadio).not.toBeDisabled();
+      // Should render both radio buttons
+      expect(cloudFormationRadio).toBeInTheDocument();
+      expect(manualRadio).toBeInTheDocument();
+    });
+
+    it('calls updatePolicy when credential type changes', () => {
+      renderWithProviders(defaultProps);
+
+      const selector = screen.getByTestId('aws-credentials-type-selector');
+
+      fireEvent.change(selector, { target: { value: 'assume_role' } });
+
+      expect(mockUpdatePolicy).toHaveBeenCalledWith({
+        updatedPolicy: expect.objectContaining({
+          inputs: expect.arrayContaining([
+            expect.objectContaining({
+              streams: expect.arrayContaining([
+                expect.objectContaining({
+                  vars: expect.objectContaining({
+                    'aws.credentials.type': { value: 'assume_role' },
+                  }),
+                }),
+              ]),
+            }),
+          ]),
+        }),
+      });
+    });
+
+    describe('Credential Type Selection', () => {
+      it('displays available credential types', () => {
+        renderWithProviders(defaultProps);
+
+        expect(screen.getByRole('option', { name: /Direct access keys/i })).toBeInTheDocument();
+        expect(screen.getByRole('option', { name: /Assume role/i })).toBeInTheDocument();
+        expect(screen.getByRole('option', { name: /Temporary keys/i })).toBeInTheDocument();
+        expect(screen.getByRole('option', { name: /Shared credentials/i })).toBeInTheDocument();
+      });
+
+      it('can change credential types', () => {
+        renderWithProviders(defaultProps);
+
+        const selector = screen.getByTestId('aws-credentials-type-selector');
+
+        // Should be able to change credential types
+        fireEvent.change(selector, { target: { value: 'assume_role' } });
+        expect(mockUpdatePolicy).toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe('Disabled State', () => {
+    it('disables radio group when disabled prop is true', () => {
+      renderWithProviders({ ...defaultProps, disabled: true });
+
+      const cloudFormationRadio = screen.getByRole('radio', { name: 'CloudFormation' });
+      const manualRadio = screen.getByRole('radio', { name: 'Manual' });
+      expect(cloudFormationRadio).toBeDisabled();
+      expect(manualRadio).toBeDisabled();
+    });
+
+    it('passes disabled prop to all interactive elements', () => {
```

Comment:
this test is exactly the same as `disables credential type selector when disabled prop is true`